flink-runtime/src/test/java/org/apache/flink/runtime/rest/handler/legacy/backpressure/BackPressureStatsTrackerImplTest.java

```diff
@@ -18,166 +18,218 @@
+	}
 
-		OperatorBackPressureStats stats = tracker.getOperatorBackPressureStats(jobVertex).get();
+	@Test
+	public void testGetOperatorBackPressureStatsAfterShutDown() throws Exception {
+		final ExecutionJobVertex jobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
 
-		// Verify the stats
-		Assert.assertEquals(sampleId, stats.getSampleId());
-		Assert.assertEquals(endTime, stats.getEndTimestamp());
-		Assert.assertEquals(taskVertices.length, stats.getNumberOfSubTasks());
+		final int requestId = 0;
+		final long startTime = System.currentTimeMillis();
+		final long endTime = startTime + 1;
+		final double backPressureRatio = 0.1;
 
-		for (int i = 0; i < taskVertices.length; i++) {
-			double ratio = stats.getBackPressureRatio(i);
-			// Traces until sub task index are back pressured
-			Assert.assertEquals((i + 1) / ((double) 4), ratio, 0.0);
-		}
+		final BackPressureStats backPressureStats = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId, startTime, endTime, backPressureRatio);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(600000, 10000, backPressureStats);
+
+		tracker.shutDown();
+
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(jobVertex);
+
+		assertFalse(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
 	}
 
-	private StackTraceElement[] createStackTrace(boolean isBackPressure) {
-		if (isBackPressure) {
-			return new StackTraceElement[] { new StackTraceElement(
-					BackPressureStatsTrackerImpl.EXPECTED_CLASS_NAME,
-					BackPressureStatsTrackerImpl.EXPECTED_METHOD_NAME,
-					"LocalBufferPool.java",
-					133) };
-		} else {
-			return Thread.currentThread().getStackTrace();
-		}
+	@Test
+	public void testStatsCleanup() throws Exception {
+		final ExecutionJobVertex jobVertex = BackPressureTrackerTestUtils.createExecutionJobVertex();
+		final int cleanUpInterval = 2000;
+		final long waitingTime = cleanUpInterval + 500;
+
+		final int requestId = 0;
+		final long startTime = System.currentTimeMillis();
+		final long endTime = startTime + 1;
+		final double backPressureRatio = 0.1;
+
+		final BackPressureStats backPressureStats = createBackPressureStats(
+			jobVertex.getTaskVertices(), requestId, startTime, endTime, backPressureRatio);
+		final BackPressureStatsTracker tracker = createBackPressureTracker(2000, 10000, backPressureStats);
+
+		// trigger back pressure stats request
+		tracker.getOperatorBackPressureStats(jobVertex);
+
+		Optional<OperatorBackPressureStats> optionalStats = tracker.getOperatorBackPressureStats(jobVertex);
+		assertTrue(optionalStats.isPresent());
+		OperatorBackPressureStats stats = optionalStats.get();
+
+		checkOperatorBackPressureStats(jobVertex.getTaskVertices(), requestId, endTime, backPressureRatio, stats);
+
+		tracker.cleanUpOperatorStatsCache();
+		assertTrue(tracker.getOperatorBackPressureStats(jobVertex).isPresent());
```

Comment:
It is very unstable to make `cleanupInterval` only 2 seconds before. If this assert is delay called 2 seconds because of gc etc, then it would cause failure here.

As mentioned above, we could move the present verify inside `checkOperatorBackPressureStats`, then every time we can also verify the internal informations not only present.

Reply:
In order to avoid this unstable issue and also avoid waiting long time below, we could break it into two tests. The first we can define long cleanup internal to verify it still present after calling cleanup. The second is defining short internal and sleeping short time to make it effect.